### ISSUE TITLE
Re-use the same npm/github token

### ISSUE THREAD

**Issue by olstenlarck** [semantic-release/cli#119]
Why not first check if some token exists in npmrc files? Then if not exist, request new one, then save it to the npmrc file, so in later calls it will resolve it instead of requesting CouchDB api again and again.

![2017-07-18-13 26 49_1280x1024_scrot](https://user-images.githubusercontent.com/5038030/28312648-88f006b0-6bbc-11e7-8253-692162d9f0a4.png)

You don't even need solutions like `keytar`, i don't see fit. The only needed thing you need to do is to request one time for tokens (no matter npm, github, travis), and then store in respective config files. And if there is answer, don't even prompt, it is slow - you should hit enter for no reason.

**Comment by tunnckoCore** [comment]
And same thing for Github

![2017-07-20-04 44 40_1280x1024_scrot](https://user-images.githubusercontent.com/5038030/28397099-263e277a-6d08-11e7-892d-44109be45735.png)

**Comment by pvdlg** [comment]
Duplicate of #143

**Comment by olstenlarck** [comment]
Mm nope. Absolutely not related.

**Comment by pvdlg** [comment]
Ok re-opening then

**Comment by gr2m** [comment]
@olstenlarck I see your point, I agree having so many tokens in GitHub/npm makes things rather noisy. What I hope we can get at least from npm very soon are tokens that are only valid for one package. I know that they are work on it, and then it all makes sense again. I’m not sure if something like this is planned for GitHub.

We do this for a few years now, but as far as I know, you’ve been the first to ask about changing that behavior. I suggest we leave the issue open for some time to see if others run into the same problem.

For the time being, maybe you can build your own CLI tool that wraps `@semantic-release/cli` which implements it the way you want? Or simply make a fork, so we can start discussing the implementation? 

I’d also love to hear thoughts by @boennemann / @christophwitzko, maybe there was a reason to implement it this way that I can’t recall

**Comment by olstenlarck** [comment]
Yea, you're right and actually I already did almost all the automation that replaces the whole semantic-release. It's pretty simple and probably feels opinionated, but exactly because it is so small and absolutely enough it's cool - `new-release`, both the npm package and the github app. The only thing that it needs is to add `NPM_TOKEN` env to the CI, nothing more - absolutely. And the best case is CircleCI v2, because it has  feature for "global context envs" and "workflows", which allows me to add NPM_TOKEN only once and to forget.

You can see the code and specific working CircleCI config at @tunnckoCore org. Also, some of the packages in that org is absolutely automated using it and that can be seen on Releases pages of these repos. It's amazing feeling to not touch anything for weeks and everything to be up to dated + released and published without any hassles :D

**Comment by gr2m** [comment]
Hey @olstenlarck glad you got it all working now for your use case 👍 Let me know if you have any further questions
